create_patched_sdk.sh

```diff
@@ -1,182 +1,236 @@
-
-    mkdir -p "$sdks_output_path_single_sdk_path"
-    cp -R "$xcode_default_sdk_path/"* "$sdks_output_path_single_sdk_path"
-
-    parse_paths=("-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/Developer"
-                 "-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/System"
-                 "-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/Library")
-
-    write_paths=("-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Developer"
-                 "-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/System"
-                 "-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Library")
-
-    "$tbd_tool" "${parse_paths[@]}" "${write_paths[@]}"
-
-    if [[ $? -ne 0 ]]; then
-        printf 'Failed to create tbds from iPhoneSimulator runtime for iOS %s\n' $xcode_sdk_ios_version
-    fi
+	if ignored $use_simulator; then
+		echo 'No DeviceSupport binaries found, falling back to dumping from simulator runtime binaries'
+	fi
+
+	sdks_output_path_single_sdk_path="$sdks_output_path/$preferred_xcode_sdk_name"
+	if [[ -d $sdks_output_path_single_sdk_path ]]; then
+		echo "SDK for $sdk_platform $xcode_sdk_version already exists"
+		exit 1
+	fi
+
+	echo "Creating sdk for $sdk_platform $xcode_sdk_version ..."
+
+	mkdir -p "$sdks_output_path_single_sdk_path"
+	cp -R "$xcode_default_sdk_path/"* "$sdks_output_path_single_sdk_path"
+
+	if [[ $sdk_platform == iOS ]]; then
+		parse_paths=("-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/Developer"
+					"-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/System"
+					"-p" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version "-r" "all" "$xcode_sim_runtime_path/Library")
+
+		write_paths=("-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Developer"
+					"-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/System"
+					"-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Library")
+
+		"$tbd_tool" "${parse_paths[@]}" "${write_paths[@]}"
+	else
+		parse_paths=("-p" "-r" "$xcode_sim_runtime_path/Developer"
+					"-p" "-r" "$xcode_sim_runtime_path/System"
+					"-p" "-r" "$xcode_sim_runtime_path/Library")
+
+		write_paths=("-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Developer"
+					"-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/System"
+					"-o" "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/Library")
+
+		"$tbd_tool" "${parse_paths[@]}" "${write_paths[@]}" $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version
+	fi
```

Comment:
I think this can all be unified into what is current labeled with "iOS" branch.
Passing `no_warnings` after `-o` is not correct, I believe. Same for `archs_option`.
When testing #68, I noticed these warnings, but didn't understand the problem.

Reply:
You got it!